helium/src/main/groovy/com/stanfy/helium/handler/codegen/java/entity/EntitiesGenerator.java

```diff
@@ -59,7 +61,19 @@ private void write(final Message type, final File classFile) {
       output = new OutputStreamWriter(new FileOutputStream(classFile), "UTF-8");
       JavaClassWriter coreWriter = Writers.pojo().create(output);
       EntitiesGeneratorOptions options = getOptions();
-      new MessageToJavaClass(options.getWriterWrapper().wrapWriter(coreWriter, options), options).write(type);
+
+      final Map<String, ClassParent> customParentMapping = options.getCustomParentMapping();
+      ClassParent classParent = null;
+      if (customParentMapping.containsKey(type.getName())) {
+        final ClassParent externalParent = customParentMapping.get(type.getName());
+        if (type.hasParent() && externalParent != null && externalParent.getExtending() != null) {
+          throw new IllegalArgumentException("Bad message type: " + type.getName()
```

Comment:
"Bad configuration for message {type}. It has a parent {type.parent}, and at the same time super class is configured for code generation ({externalParent})".